components/Textarea/Textarea.tsx

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Gridiron Survivor.
+// Licensed under the MIT License.
+
+'use client';
+import { cn } from '@/utils/utils';
+import { useImperativeHandle } from 'react';
+import * as React from 'react';
+
+interface UseAutosizeTextAreaProps {
+  textAreaRef: HTMLTextAreaElement | null;
+  minHeight?: number;
+  maxHeight?: number;
+  triggerAutoSize: string;
+}
+
+/**
+ * useAutosizeTextArea hook
+ * @param {UseAutosizeTextAreaProps} props - The props
+ */
+export const useAutosizeTextArea = ({
+  textAreaRef,
+  triggerAutoSize,
+  maxHeight = Number.MAX_SAFE_INTEGER,
+  minHeight = 0,
+}: UseAutosizeTextAreaProps): void => {
+  const [init, setInit] = React.useState(true);
+  React.useEffect(() => {
+    const offsetBorder = 2;
+    if (textAreaRef) {
+      if (init) {
+        textAreaRef.style.minHeight = `${minHeight + offsetBorder}px`;
+        if (maxHeight > minHeight) {
+          textAreaRef.style.maxHeight = `${maxHeight}px`;
+        }
+        setInit(false);
+      }
+      textAreaRef.style.height = `${minHeight + offsetBorder}px`;
+      const scrollHeight = textAreaRef.scrollHeight;
+      if (scrollHeight > maxHeight) {
+        textAreaRef.style.height = `${maxHeight}px`;
+      } else {
+        textAreaRef.style.height = `${scrollHeight + offsetBorder}px`;
+      }
+    }
+  }, [textAreaRef, triggerAutoSize]);
```

Comment:
This seems very cluttered, is there a way you can add new lines to make it easier to read?

Reply:
I believe they will be removed by the formatter but I can try.